### PR TITLE
Extract admin#components to a separate controller

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/controllers/course/admin/component_settings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Course::Admin::ComponentSettingsController < Course::Admin::Controller
+  before_action :load_settings, only: [:edit, :update]
```
Comment (lowjoel) [inline app/controllers/course/admin/component_settings_controller.rb]: No need `only:`

File: spec/features/course/admin/component_settings_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-RSpec.feature 'Course: Administration: Components' do
+RSpec.feature 'Course: Administration: Component' do
```
Comment (lowjoel) [inline spec/features/course/admin/component_settings_spec.rb]: Components is correct.

Comment (allenwq) [inline spec/features/course/admin/component_settings_spec.rb]: How about routes ? 

I'm using `course/:id/component` now

Comment (lowjoel) [inline spec/features/course/admin/component_settings_spec.rb]: plural also.

**Pull request merged**